main.go

```diff
@@ -150,27 +195,184 @@ func readConfig(filename string) (*config, error) {
 	return &c, nil
 }
 
+// myClain define token struct
 type myClaim struct {
 	jwt.Claims
 	Roles []string `json:"roles,omitempty"`
 }
 
-func validateClaims(t string, sufficientRoles []string) error {
+// validateClaims check role fom config sufficetRoles is in token roles claim
+func validateClaims(t string, sufficientRoles []string, username string, addGroup bool, addmembership bool) error {
 	token, err := jwt.ParseSigned(t)
 	if err != nil {
 		return fmt.Errorf("error parsing token: %w", err)
 	}
 
 	claims := myClaim{}
 	if err := token.UnsafeClaimsWithoutVerification(&claims); err != nil {
-		return fmt.Errorf("unable to extract claims from token:%w", err)
+		return fmt.Errorf("unable to extract claims from token: %w", err)
 	}
 	for _, role := range claims.Roles {
+
+		if addGroup {
+			createGroup(role, username)
+		}
+
+		if addmembership {
+			addUserToGroup(role, username)
+		}
+
 		for _, sr := range sufficientRoles {
 			if role == sr {
+				log.Print("validateClaims access granted role " + role + " is in token")
 				return nil
 			}
 		}
 	}
-	return fmt.Errorf("role:%s not found", sufficientRoles)
+	return fmt.Errorf("role: %s not found", sufficientRoles)
+}
+
+// createUser this create user is not exsits
+func createUser(username string) {
+	_, err := user.LookupUser(username)
```

Comment:
https://pkg.go.dev/os/user#Lookup is part of stdlib, please prefer these !

Reply:
use stdlib right now see last commit